GNUmakefile

```diff
@@ -46,66 +66,135 @@ endif
 export PYTHON_BIN := python$(PYTHON_VERSION)
 
 ifeq ($(VERSION),)
-export VERSION := $(shell python3 -m setuptools_scm | tr -s '-' '~' | sed 's/^v//')
+export VERSION := $(shell python3 -m setuptools_scm 2>/dev/null | tr -s '-' '~' | tr -s '+' '_' | sed 's/^v//')
 endif
 
-# Might want to run with parallelism by default to make sure people don't
-# introduce dependencies in tests and with randomness.
-# SHELLSEPCARGS:=--jobs 4
-SHELLSPECARGS:=
-ENTR:=entr
-SH:=sh
-BASH:=bash
-KSH:=ksh
-ZSH:=zsh
-SHELLS:=$(SH) $(BASH) $(KSH) $(ZSH)
-DATE:=date
-
-SPEC_FILE ?= ${NAME}.spec
-SOURCE_NAME ?= ${NAME}
-BUILD_DIR ?= $(PWD)/dist/rpmbuild
-SOURCE_PATH := ${BUILD_DIR}/SOURCES/${SOURCE_NAME}-${VERSION}.tar.bz2
-
-phonies:=ci
-phonies+=test
-phonies+=test-all
+ifeq ($(VERSION),)
+$(warning VERSION not set! Verify setuptools_scm[toml] is installed and try again.)
+endif
 
-.PHONY: $(phonies)
+#############################################################################
+# Post Release handling
+# "post" releases are useful for when non-code changes are made after
+# a release was created:
+# - When a README is updated, or CHANGE_LOG after a release was made
+# - When build changes occur for distributing the application to another
+#	platform but the code has zero changes
+#############################################################################
+
+# NOTE: 1.0.0 and 1.0.0.post0 mean the same thing, so to keep things simple if a post0 stable tag is detected it
+#		will be truncated.
+export VERSION := $(shell echo $(VERSION) | sed -E 's/\.post0$$.*//')
```

Comment:
We could probably yeet all this stuff into .mk files and make... make functions out of em so we can reuse the same thing elsewhere. Given this is all library and makefiles are kinda like shell in that regard.